core/dovecot/start.py

```diff
@@ -21,11 +21,20 @@ def start_podop():
     ])
 
 # Actual startup script
-os.environ["FRONT_ADDRESS"] = system.resolve_address(os.environ.get("HOST_FRONT", "front"))
-os.environ["REDIS_ADDRESS"] = system.resolve_address(os.environ.get("HOST_REDIS", "redis"))
-os.environ["ADMIN_ADDRESS"] = system.resolve_address(os.environ.get("HOST_ADMIN", "admin"))
-if os.environ["WEBMAIL"] != "none":
-    os.environ["WEBMAIL_ADDRESS"] = system.resolve_address(os.environ.get("HOST_WEBMAIL", "webmail"))
+
+# set default values
+os.environ["FRONT_ADDRESS"] = os.environ.get("HOST_FRONT", "front")
+os.environ["REDIS_ADDRESS"] = os.environ.get("HOST_REDIS", "redis")
+os.environ["ADMIN_ADDRESS"] = os.environ.get("HOST_ADMIN", "admin")
+os.environ["WEBMAIL_ADDRESS"] = os.environ.get("HOST_WEBMAIL", "webmail")
+
+# resolve hosts to ip addresses
+if os.environ.get('RESOLVE_SERVICE_HOSTNAMES','true').lower() in ('true','yes'):
+    os.environ["FRONT_ADDRESS"] = system.resolve_address(os.environ.get("FRONT_ADDRESS"))
+    os.environ["REDIS_ADDRESS"] = system.resolve_address(os.environ.get("REDIS_ADDRESS"))
+    os.environ["ADMIN_ADDRESS"] = system.resolve_address(os.environ.get("ADMIN_ADDRESS"))
+    if os.environ["WEBMAIL"] != "none":
+        os.environ["WEBMAIL_ADDRESS"] = system.resolve_address(os.environ.get("WEBMAIL_ADDRESS"))
 
```

Comment:
I'm not sure what the logic is here? In #940 we tried to achieve some consistency:
- `HOST_*` caries unresolved host names with optional port suffix
- `*_ADDRESS` caries resolved IP addresses with optional port suffix

In the above code I'm reading: 
1. Assign the unresolved hostname into (global) address.
2. if `RESOLVE_SERVICE_HOSTNAMES` is `true` (default) or `yes`:
2.1. Overwrite address with a resolved address

Perhaps something like:
````
os.environ.["FRONT_ADDRESS"] = os.environ.get("FRONT_ADDRESS",  system.resolve_address(os.environ.get("HOST_FRONT", "front"))
````
Using this logic, 
1. If the user sets `FRONT_ADDRESS` it is used without resolving
2. If `FRONT_ADDRESS` is emtpy, the resolver gets called:
2.1. If `HOST_FRONT` is set, that host name will be resolved
2.2. If `HOST_FRONT` is empty, the default host name `front` will be resolved.

This way we don't have to mangle between variables, or declare a new variable like `RESOLVE_SERVICE_HOSTNAMES`

Reply:
I have changed it accordingly